src/utils/trie.ts

```diff
@@ -79,9 +79,21 @@ export class TrieNode<T> {
             prefix: alias.replace(/\*$/, ''),
             // Normalize paths.
             paths: paths[alias].map((path) => {
-              path = path
-                .replace(/\*$/, '')
-                .replace(/\.([mc])?ts(x)?$/, '.$1js$2');
+              path = path.replace(/\*$/, '');
+
+              const { dir, base } = parse(path);
+
+              const dotIndex = base.indexOf('.');
+              const name = base.slice(0, dotIndex);
+              const extension = base.slice(dotIndex);
+
+              const normalizedExtension = extension.replace(
+                /^\.([mc])?ts(x)?$/,
```

Comment:
I added a `^` which means the whole extension must be matched. This may need to have smarter logic.

Files to definitely exclude:
- `foo.d.css.ts` - The correct way of defining a declaration file for `foo.css` file. Cannot point to a `js` file.
- `foo.css.d.ts` - A legacy way of defining a `foo.css` file. Cannot point to a `js` file.
- `foo.d.ts` - Declaration files for a JS file. This JS file could technically exist in repo but if it does the `.d.ts` file should probably exist alongside. Also if you're doing this you probably should have a plain `.ts` file instead.
- `foo.xyz.ts` - I'm unsure why you'd write this but it would be broken by this regex change. Maybe some libraries make this common or some repos do this?

If you'd like me to look into a better way, I can.

Reply:
I decided I may as well push a fix for `foo.xyz.ts`. See "Add explicit isDTS detection"